examples/ErrorHandling/HandlePartialFailures.cs

```diff
@@ -0,0 +1,166 @@
+    /// </summary>
+    public class HandlePartialFailures : ExampleBase
+    {
+        /// <summary>
+        /// Main method, to run this code example as a standalone application.
+        /// </summary>
+        /// <param name="args">The command line arguments.</param>
+        public static void Main(string[] args)
+        {
+            HandlePartialFailures codeExample = new HandlePartialFailures();
+            Console.WriteLine(codeExample.Description);
+            try
+            {
+                long campaignId = long.Parse("INSERT_CAMPAIGN_ID_HERE");
+
+                // The Google Ads customer ID for which the call is made.
+                long customerId = long.Parse("INSERT_CUSTOMER_ID_HERE");
+
+                codeExample.Run(new GoogleAdsClient(), customerId, campaignId);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("An exception occurred while running this code example. {0}",
+                    ExampleUtilities.FormatException(e));
+            }
+        }
+
+        /// <summary>
+        /// Returns a description about the code example.
+        /// </summary>
+        public override string Description
+        {
+            get
+            {
+                return "This code example demonstrates how to handle partial failures.";
+            }
+        }
+
+        /// <summary>
+        /// Runs the code example.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The customer ID for which the call is made.</param>
+        /// <param name="campaignId">Id of the campaign to which ad groups are added.</param>
+        public void Run(GoogleAdsClient client, long customerId, long campaignId)
+        {
+            // Get the AdGroupServiceClient.
+            AdGroupServiceClient adGroupService = client.GetService(Services.V1.AdGroupService);
+
+
+            // This ad group should be created successfully - assuming the campaign in the params
+            // exists.
+            AdGroup adgroup1 = new AdGroup()
+            {
+                Campaign = ResourceNames.Campaign(customerId, campaignId),
+                Name = "Valid AdGroup: " + ExampleUtilities.GetRandomString()
+            };
+
+            // This ad group will always fail - campaign ID 0 in resource names is never valid.
+            AdGroup adgroup2 = new AdGroup()
+            {
+                Campaign = ResourceNames.Campaign(customerId, 0),
+                Name = "Broken AdGroup: " + ExampleUtilities.GetRandomString()
+            };
+
+            // This ad group will always fail - duplicate ad group names are not allowed.
+            AdGroup adgroup3 = new AdGroup()
+            {
+                Campaign = ResourceNames.Campaign(customerId, 0),
+                Name = adgroup1.Name
+            };
+
+            // Create the operations.
+            AdGroupOperation op1 = new AdGroupOperation()
```

Comment:
Just wondering, is there a shorter way to structure these blocks? It seems to take quite a lot of vertical space to create the operations...

Reply:
Done.